### PR TITLE
Feature/resume parser 2

### PR DESCRIPTION
**Pull request by krsween**
### Release Notes:

1. Added LinkedIn Partner Verification
2. Added full resume parsing
3. Modified i18n of months

### CODE REVIEW

File: src/app/modal/modal.controller.js [hunk 1]
```diff
@@ -151,36 +140,102 @@ class CareerPortalModalController {
...
+            }
+            // Add line break
+            this.linkedInData.resume += hardBreak;
+        }
+
         // Work Experience Block
-        this.linkedInData.resume = this.$filter('i18n')('modal.workExperience') + lineBreak;
+        this.linkedInData.resume += this.$filter('i18n')('modal.workExperience') + lineBreak;
         // Positions
         if (userProfile.positions) {
```
Comment (TheReal) [inline src/app/modal/modal.controller.js]: Consider factoring this the same way as education values array (line 146):

```
if (userProfile.positions && userProfile.positions.values) {
    for (var positions = userProfile.positions.values, var i = 0, var c = positions.length; i < c; i++) {
        ...
    }
}
```

File: src/app/modal/modal.controller.js [hunk 2]
```diff
@@ -151,36 +140,102 @@ class CareerPortalModalController {
...
-                    this.linkedInData.resume += months[userProfile.positions.values[0].endDate.month - 1] + ' ' + userProfile.positions.values[0].endDate.year || '';
-                } else {
-                    if (userProfile.positions.values[0].isCurrent) {
-                        this.linkedInData.resume += this.$filter('i18n')('modal.present');
+            var positions = userProfile.positions.values;
+            // Iterate through each position
+            if (positions && positions.length) {
+                for (var ii = 0; ii < positions.length; ii++) {
+                    // Add Employee section header
+                    this.linkedInData.resume += (positions[ii].company.name || '') + ' ';
```
Comment (TheReal) [inline src/app/modal/modal.controller.js]: Is the company field on the positions anonymous object guaranteed to by defined and non-null? Consider a defensive de-referencing expression:

`((positions[i].company || {}).name || ' ') + ' '`

File: src/app/modal/modal.controller.js [hunk 3]
```diff
@@ -151,36 +140,102 @@ class CareerPortalModalController {
...
+                    // Start Date
+                    if (positions[ii].startDate) {
+                        this.linkedInData.resume += months[positions[ii].startDate.month - 1] + ' ' + positions[ii].startDate.year + ' - ' || '';
```
Comment (TheReal) [inline src/app/modal/modal.controller.js]: Same as above comment about defined and non-null.

File: src/app/modal/modal.controller.js [hunk 4]
```diff
@@ -151,36 +140,102 @@ class CareerPortalModalController {
...
+                    } else {
+                        if (positions[ii].isCurrent) { // jshint ignore:line
+                            this.linkedInData.resume += this.$filter('i18n')('modal.present');
+                        }
+                    }
+                    this.linkedInData.resume += lineBreak;
+                    // Title
+                    this.linkedInData.resume += positions[ii].title + lineBreak || '';
+                    // Industry
+                    this.linkedInData.resume += positions[ii].company.industry ? positions[ii].company.industry + lineBreak : '';
```
Comment (TheReal) [inline src/app/modal/modal.controller.js]: Same as company comment above.

File: src/app/modal/modal.controller.js [hunk 5]
```diff
@@ -151,36 +140,102 @@ class CareerPortalModalController {
...
-                // Industry
-                this.linkedInData.resume += userProfile.positions.values[0].company.industry ? userProfile.positions.values[0].company.industry + lineBreak : '';
-                if (userProfile.positions.values[0].location) {
-                    // Locale
-                    this.linkedInData.resume += userProfile.positions.values[0].location.name + lineBreak || '';
+            }
+        }
+
+        // Skills
+        if (userProfile.skills && userProfile.skills.values) {
```
Comment (TheReal) [inline src/app/modal/modal.controller.js]: Consider following same value-check and iteration code as decided for the above.

**Comment by TheReal** [comment]
Assigning back to Kameron for consideration of comments.

**Comment by jgodi** [comment]
Agree with @TheReal's comments, otherwise LGTM

File: src/app/modal/modal.controller.js (hunk 4 above)
Comment (krsween) [inline src/app/modal/modal.controller.js]: In the tests I've run, if there's a startDate object, there's always a month field, That field may resolve as `''`, though.

File: src/app/modal/modal.controller.js (hunk 2 above)
Comment (krsween) [inline src/app/modal/modal.controller.js]: Yeah, this one could definitely blow up.

**Pull request merged**